pkg/sessions/tests/session_store_tests.go

```diff
@@ -0,0 +1,443 @@
+				broken := "BrokenSessionFromADifferentSessionImplementation"
+				value := encryption.SignedValue(in.cookieOpts.Secret, in.cookieOpts.Name, []byte(broken), time.Now())
+				cookie := cookiesapi.MakeCookieFromOptions(in.request, in.cookieOpts.Name, value, in.cookieOpts, in.cookieOpts.Expire, time.Now())
+				in.request.AddCookie(cookie)
+
+				err := in.ss().Save(in.response, in.request, in.session)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It("sets a `set-cookie` header in the response", func() {
+				Expect(in.response.Header().Get("set-cookie")).ToNot(BeEmpty())
+			})
+
+			It("Ensures the session CreatedAt is not zero", func() {
+				Expect(in.session.CreatedAt.IsZero()).To(BeFalse())
+			})
+
+			CheckCookieOptions(in)
+		})
+
+		Context("with an expired saved session", func() {
+			var err error
+			BeforeEach(func() {
+				By("saving a session")
+				req := httptest.NewRequest("GET", "http://example.com/", nil)
+				saveResp := httptest.NewRecorder()
+				err = in.ss().Save(saveResp, req, in.session)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("and clearing the session")
+				for _, c := range saveResp.Result().Cookies() {
+					in.request.AddCookie(c)
+				}
+				clearResp := httptest.NewRecorder()
+				err = in.ss().Clear(clearResp, in.request)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("then saving a request with the cleared session")
+				err = in.ss().Save(in.response, in.request, in.session)
+			})
+
+			It("no error should occur", func() {
+				Expect(err).ToNot(HaveOccurred())
+			})
+		})
+	})
+
+	Context("when Clear is called", func() {
+		BeforeEach(func() {
+			req := httptest.NewRequest("GET", "http://example.com/", nil)
+			saveResp := httptest.NewRecorder()
+			err := in.ss().Save(saveResp, req, in.session)
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, c := range saveResp.Result().Cookies() {
+				in.request.AddCookie(c)
+			}
+			err = in.ss().Clear(in.response, in.request)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("sets a `set-cookie` header in the response", func() {
+			Expect(in.response.Header().Get("Set-Cookie")).ToNot(BeEmpty())
+		})
+
+		CheckCookieOptions(in)
+	})
+
+	Context("when Load is called", func() {
+		BeforeEach(func() {
+			req := httptest.NewRequest("GET", "http://example.com/", nil)
+			resp := httptest.NewRecorder()
+			err := in.ss().Save(resp, req, in.session)
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, cookie := range resp.Result().Cookies() {
+				in.request.AddCookie(cookie)
+			}
+		})
+
+		Context("before the refresh period", func() {
+			LoadSessionTests(in)
+		})
+
+	})
+}
+
+func LoadSessionTests(in *testInput) {
+	var loadedSession *sessionsapi.SessionState
+	BeforeEach(func() {
+		var err error
+		loadedSession, err = in.ss().Load(in.request)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("loads a session equal to the original session", func() {
+		if in.cookieOpts.Secret == "" {
```

Comment:
We should clean this up now that valid length secrets and a corresponding cipher is always requires in validation.